src/System.Reflection.Emit/tests/ParameterBuilder/ParameterBuilderSetConstant.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    public class ParameterBuilderSetConstant
+    {
+        public static IEnumerable<object[]> SetConstant_ReferenceTypes_TestData()
+        {
+            yield return new object[] { typeof(object) };
+            yield return new object[] { typeof(string) };
+            yield return new object[] { typeof(UserDefinedClass) };
+        }
+
+        public static IEnumerable<object[]> SetConstant_NullableValueTypes_TestData()
+        {
+            yield return new object[] { typeof(bool?) };
+            yield return new object[] { typeof(byte?) };
+            yield return new object[] { typeof(char?) };
+            yield return new object[] { typeof(DateTime?) };
+            yield return new object[] { typeof(decimal?) };
+            yield return new object[] { typeof(double?) };
+            yield return new object[] { typeof(float?) };
+            yield return new object[] { typeof(int?) };
+            yield return new object[] { typeof(long?) };
+            yield return new object[] { typeof(sbyte?) };
+            yield return new object[] { typeof(short?) };
+            yield return new object[] { typeof(uint?) };
+            yield return new object[] { typeof(ulong?) };
+            yield return new object[] { typeof(UserDefinedStruct?) };
+            yield return new object[] { typeof(ushort?) };
+        }
+
+        public static IEnumerable<object[]> SetConstant_ValueTypes_TestData()
+        {
+            yield return new object[] { typeof(bool) };
+            yield return new object[] { typeof(byte) };
+            yield return new object[] { typeof(char) };
+            yield return new object[] { typeof(DateTime) };
+            yield return new object[] { typeof(decimal) };
+            yield return new object[] { typeof(double) };
+            yield return new object[] { typeof(float) };
+            yield return new object[] { typeof(int) };
+            yield return new object[] { typeof(long) };
+            yield return new object[] { typeof(sbyte) };
+            yield return new object[] { typeof(short) };
+            yield return new object[] { typeof(uint) };
+            yield return new object[] { typeof(ulong) };
+            yield return new object[] { typeof(UserDefinedStruct) };
+            yield return new object[] { typeof(ushort) };
+        }
+
+        [Theory]
+        [MemberData(nameof(SetConstant_ReferenceTypes_TestData))]
+        [MemberData(nameof(SetConstant_NullableValueTypes_TestData))]
```

Comment:
The valuetype cases will fail on the NETFX (.NET 4) platforms. You'll want to separate those out into its own Theory method and add the custom attribute:

`[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Passing null for SetConstant on value types not supported on NETFX")]
`

Reply:
Done.